packages/syft/src/syft/custom_worker/builder.py

```diff
@@ -0,0 +1,98 @@
+# stdlib
+import os.path
+from pathlib import Path
+from typing import Tuple
+
+# third party
+import docker
+
+# relative
+from .config import CustomWorkerConfig
+
+
+class CustomWorkerBuilder:
+    TYPE_CPU = "cpu"
+    TYPE_GPU = "gpu"
+
+    DOCKERFILE_PROD_PATH = os.path.expandvars("$APPDIR/grid/")
+    DOCKERFILE_DEV_PATH = "../../../../../grid/backend/"
+
+    CUSTOM_IMAGE_PREFIX = "custom-worker"
+
+    BUILD_MAX_WAIT = 30 * 60
+
+    def build_image(self, config: CustomWorkerConfig) -> None:
+        """
+        Builds a Docker image for the custom worker based on the provided configuration.
+        Args:
+            config (CustomImageConfig): The configuration for building the Docker image.
+        Returns:
+            bool: True if the image was built successfully, raises Exception otherwise.
+        """
+
+        # remove once GPU is supported
+        if config.build.gpu:
+            raise Exception("GPU custom worker is not supported yet")
+
+        type = self.TYPE_GPU if config.build.gpu else self.TYPE_CPU
+
+        contextdir, dockerfile = self.find_worker_ctx(type)
+
+        imgtag = config.get_signature()[:8]
+
+        build_args = {
+            "PYTHON_VERSION": config.build.python_version,
+            "SYSTEM_PACKAGES": config.build.merged_system_pkgs(),
+            "PIP_PACKAGES": config.build.merged_python_pkgs(),
+            "CUSTOM_CMD": config.build.merged_custom_cmds(),
+        }
+
+        print(
+            f"Building dockerfile={dockerfile} "
+            f"in context={contextdir} "
+            f"with args={build_args}"
+        )
+
+        try:
+            client = docker.from_env()
+
+            # TODO: Push logs to mongo/seaweed?
+            client.images.build(
+                path=str(contextdir),
+                dockerfile=dockerfile,
+                pull=True,
+                tag=f"{self.CUSTOM_IMAGE_PREFIX}-{type}:{imgtag}",
+                timeout=self.BUILD_MAX_WAIT,
+                buildargs=build_args,
+            )
+
+            return
+        except docker.errors.BuildError as e:
+            raise e
+        except docker.errors.APIError as e:
+            raise e
+        except Exception as e:
+            raise e
```

Comment:
Doesn't 

```py
except Exception as e:
    raise e
```

already catches all of these?

Reply:
i'm keeping them here so that we know what kind of errors we want to handle when integrating with worker service. This will eventually get streamlined...